app/controllers/developers_controller.rb

```diff
@@ -0,0 +1,66 @@
+class DevelopersController < ApplicationController
+  skip_before_action :ensure_current_user
+
+
+  def index
+    @developers = Developer.all
+    # NEXT STEP - order by rating
+  end
+
+  def show
+    @developer = Developer.find_by(id: params[:id])
+  end
+
+  def new
+    session[:registration_type] = 'Developer'
+    @developer = Developer.new
+  end
+
+  def create
+    @developer = Developer.new(developer_params)
+    if @developer.save
+      flash.notice = "You're in! Welcome."
+      session[:user_id] = @developer.id
+      redirect_to edit_developer_path(@developer)
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @developer = Developer.find_by(id: params[:id])
+  end
+
+  def update
+    @developer = Developer.find_by(id: params[:id])
+    @skills = params[:skills].split(",").map {|skill| skill.strip}
+    if @developer.update(developer_edit_params)
+        @skills.each do |skill|
+        @developer.skills << Skill.find_or_create_by(name: skill)
```

Comment:
I'd maybe put the find_or_create_by stuff in the model - so that the model goes through skills and does what's required to create new records.